### PR TITLE
[hive]hive create table should force primary key not null

### PR DESCRIPTION
**Pull request by Alibaba-HZY**
hive create table should force primary key not null

### Tests

HiveCatalogITCaseBase

### API and Format

no

### Documentation

no

### CODE REVIEW

File: paimon-core/src/main/java/org/apache/paimon/schema/SchemaManager.java [hunk 1]
```diff
@@ -146,7 +146,23 @@ public TableSchema createTable(Schema schema) throws Exception {
                 }
                 options.remove(CoreOptions.PRIMARY_KEY.key());
             }
-
+            List<String> finalPrimaryKeys = primaryKeys;
+            fields =
```
Comment (JingsongLi) [inline paimon-core/src/main/java/org/apache/paimon/schema/SchemaManager.java:150]: Actually, I don't get why.

`Schema.normalizeFields` will convert pk fields to not-null.

Comment (Alibaba-HZY) [reply paimon-core/src/main/java/org/apache/paimon/schema/SchemaManager.java:150]: when hive create table primary key  is created by 
image
is not from Schema.normalizeFields

Comment (JingsongLi) [reply paimon-core/src/main/java/org/apache/paimon/schema/SchemaManager.java:150]: I see, maybe we should move the logical from `SchemaManager` to `Schema`.

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**